os/arch/arm/src/amebasmart/amebasmart_usb.c

```diff
@@ -761,29 +761,35 @@ int amebasmart_up_usbinitialize(struct amebasmart_usbdev_s *priv)
 /*It takes sometime for USB initialization to complete, so during boot up stage, it /dev/console will be
  register with serial first to prevent opening an empty path, then once usb is ready, re-register it to usb and
  open the fd*/
+extern int tash_start(void);
+extern void tash_stop(void);
+static struct task_struct register_usb_task;
 static int register_usb(void)
 {
 	while(!cdc_acm_ready_flag){
-		usleep(1000);
+		usleep(100);
 	}
 	USB_DEV.isconsole = true;
 	uart_register("/dev/ttyACM0", &USB_DEV);
 	/*unregister /dev/console because serial was sharing the same fd during serial initialization*/
 	unregister_driver("/dev/console");
+	tash_stop();
```

Comment:
You should not call the apl API at kernel side. And USB BSP should not controll the TASH. Please find another way.

Reply:
Dear @sunghan-chang,

In my understanding, during the initalization of the TASH task, it opened the path /dev/console, and this fd will be used in a do while loop until TASH stopped. So during boot up stage, if USBD is not ready, /dev/console will be registered to loguart and logs and input will be direct to loguart in TASH task. Even if i re-register it to USB (when USB device is connected,) the logs will not redirect to usb unless i can close the fd and re-open it. 

Is it possible to change tash_main.c file to add flag check for usb device?

Reply:
Hello @zhongnuo-tang .
Regardless of USB connection, when we enable USB console, then logs should be out through USB only. We should not redirect it to UART. We enable USB console due to performance issue with UART. If we redirect logs to UART according to USB connection, we don't need to enable USB console. (Almost products has no USB connection)

Let's make it as UART does. Even UART is not connected, it does not block Tizen Lite booting. They should be independent. ("They" are booting and UART connection in physical)

Reply:
Dear @sunghan-chang,
Understood, can i assume that all the logs during USB disconnection period can be ignored/loss? Initially the busy-wait was to ensure that USB logging should only start working when USB device is connected and ready.

Reply:
In UART, it's loss, even uart sends logs but there is no receiver.
Anything is ok. BUT we have the log dump module, it saves logs regardless of UART or USB connection. It should be working.